core/core/xchaincore.go

```diff
@@ -1033,26 +1033,30 @@ func (xc *XChainCore) GetBlockChainStatus(in *pb.BCStatus) *pb.BCStatus {
 	}
 
 	meta := xc.Ledger.GetMeta()
-	out.Meta = meta
-	utxoMeta := xc.Utxovm.GetMeta()
-	out.UtxoMeta = utxoMeta
-
+	if viewOption == pb.ViewOption_NONE || viewOption == pb.ViewOption_LEDGER || viewOption == pb.ViewOption_BRANCHINFO {
+		out.Meta = meta
+	}
+	if viewOption == pb.ViewOption_NONE || viewOption == pb.ViewOption_UTXOINFO {
+		utxoMeta := xc.Utxovm.GetMeta()
+		out.UtxoMeta = utxoMeta
+	}
 	ib, err := xc.Ledger.QueryBlock(meta.TipBlockid)
 	if err != nil {
 		out.Header.Error = HandlerLedgerError(err)
 		return out
 	}
 	out.Block = ib
```

Comment:
Do not return out.Block if option is not None.

Reply:
Fixed, thanks